### PR TITLE
BO-82 [socket-redis] add token authentication for status end-point

### PR DESCRIPTION
**Pull request by kris-lab**
Part of https://github.com/cargomedia/puppet-cargomedia/issues/1702
Depends on https://github.com/cargomedia/socket-redis/pull/97

### CODE REVIEW

**Comment by kris-lab** [comment]
@tomaszdurka @vogdb please have a look

**Comment by vogdb** [comment]
looks fine to me. Travis failed somehow

File: lib/bipbip/version.rb [hunk 1]
```diff
@@ -1,3 +1,3 @@
 module Bipbip
-  VERSION = '0.7.11'.freeze
+  VERSION = '0.7.12'.freeze
```
Comment (tomaszdurka) [inline lib/bipbip/version.rb]: Well that should be rather minor! :)

File: lib/bipbip/plugin/socket_redis.rb [hunk 2]
```diff
@@ -23,7 +23,13 @@ def fetch_socket_redis_status
       url = config['url'] || 'http://localhost:8085/status'
       uri = URI.parse(url)
 
-      response = Net::HTTP.get_response(uri)
+      request = Net::HTTP::Get.new(uri)
+      request['authorization'] = "token #{config['status_token']}"
```
Comment (tomaszdurka) [inline lib/bipbip/plugin/socket_redis.rb]: I would not expect that's how you add headers to request object. But seems right as here https://ruby-doc.org/stdlib-2.4.0/libdoc/net/http/rdoc/Net/HTTP.html#class-Net::HTTP-label-Setting+Headers

**Comment by tomaszdurka** [comment]
@kris-lab lgtm please let me know in https://github.com/cargomedia/puppet-cargomedia/issues/1702 when bipbip is deployed with sending header. The plan is to release socket-redis once clients are ready.

**Comment by kris-lab** [comment]
@tomaszdurka there are also tickets for `puppet` modules before we can continue. Please review them.

**Comment by tomaszdurka** [comment]
As wrote I think we should rather finish up clients first then. Sending the header makes no harm!

File: lib/bipbip/plugin/socket_redis.rb (hunk 2 above)
Comment (kris-lab) [inline lib/bipbip/plugin/socket_redis.rb]: Maybe we should not set this `authorization` header if `config['status_token']` is undef, wdyt?

**Comment by kris-lab** [comment]
@tomaszdurka please have a look. I am gonna released this soon.

**Comment by kris-lab** [comment]
As implemented in [socket-redis](https://github.com/cargomedia/socket-redis/blob/master/lib/status-server.js#L110) the `status-secret` is optional.

**Pull request merged**